src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -66,12 +76,24 @@ export function kubeNodes(): V1NetworkPolicyPeer[] {
 }
 
 /**
- * When a node is created or updated, if it's Ready, add its IP to the set,
+ * When a node is created or updated, if it's Ready, add its IP to the nodeSet and nodeNameToIPMap,
  * rebuild the policies, and update the NetworkPolicies.
  */
 export async function updateKubeNodesFromCreateUpdate(node: kind.Node) {
   const ip = getNodeInternalIP(node);
-  if (ip) nodeSet.add(ip);
+  const nodeName = node.metadata?.name;
+
+  if (ip && nodeName) {
```

Comment:
Related to my other comment - maybe the easiest approach is just to use this same conditional in `initAllNodesTarget`? Currently on initialization we add all nodes to the set (regardless of name), but during new creations/updates we only add them if a name is present.